### ISSUE TITLE
loop: implement missing functionality

### ISSUE THREAD

**Issue by smathot** [shyras/osweb#129]
The following `loop` functionality should still be implemented:

- Resume after break
- Evaluate on first cycle
- File source

**Comment by dschreij** [comment]
File source will be ambitious because of the security limitations JavaScript poses concerning the loading of external files. To be honest, I have never used the Resume after break feature. Could you provide a short description of what it is supposed to do (or point me to the doc page that I can't seem to find on my own)?

**Comment by smathot** [comment]
> File source will be ambitious because of the security limitations JavaScript poses concerning the loading of external files.

It strikes me that this is not fundamentally different from the script being read and the files in the pool being extracted. Or am I missing something?

> To be honest, I have never used the Resume after break feature. Could you provide a short description of what it is supposed to do (or point me to the doc page that I can't seem to find on my own)?

Good point! I'll update the documentation. But basically, it means that when the `loop` stops and not all cycles have been executed, then the next time that the `loop` is run, it resumes where it left of. Like a generator, in a way. Does that make sense?

**Comment by dschreij** [comment]
> It strikes me that this is not fundamentally different from the script being read and the files in the pool being extracted. Or am I missing something?

That's true. I forgot about that. The file should be accessible through an URL to the server on which it resides (so no need to access the local filesystem).

And the second example indeed makes sense. Never a shortage of challenges with you... ;)
